framework/Samplers/Importance.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  This module contains the Umbrella sampling strategy
+
+  Created on Feb 15, 2021
+  @author: Tanaya
+  supercedes Samplers.py from crisr
+"""
+#for future compatibility with Python 3--------------------------------------------------------------
+from __future__ import division, print_function, unicode_literals, absolute_import
+#End compatibility block for Python 3----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+import numpy as np
+from operator import mul
+from functools import reduce
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from .ForwardSampler import ForwardSampler
+from utils import utils,randomUtils,InputData, InputTypes
+#Internal Modules End--------------------------------------------------------------------------------
+
+class Importance(ForwardSampler):
+  """
+    MONTE CARLO Sampler
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(Importance, cls).getInputSpecification()
+    samplerInitInput = InputData.parameterInputFactory("samplerInit")
+    limit = InputData.parameterInputFactory("limit", contentType=InputTypes.IntegerType)
+    samplerInitInput.addSub(limit)
+    targetDistribution = InputData.parameterInputFactory("distribution")
+    samplerInitInput.addSub(targetDistribution)
+    samplingTypeInput = InputData.parameterInputFactory("samplingType", contentType=InputTypes.StringType)
+    samplerInitInput.addSub(samplingTypeInput)
```

Comment:
It's not a problem either way, but is there a reason you want to put the sampling type and distribution information inside the samplerInit rather than parallel with it?

Again, fine either way.